test/e2e/test_override_job_user.py

```diff
@@ -234,3 +237,209 @@ def test_env_var_user_override(
+                        },
+                        "script": {
+                            "embeddedFiles": [
+                                {
+                                    "name": "whoami",
+                                    "type": "TEXT",
+                                    "runnable": True,
+                                    "data": "\n".join(
+                                        [
+                                            "#!/bin/bash",
+                                            'echo "I am: $(whoami)"',
+                                        ]
+                                    ),
+                                },
+                            ],
+                            "actions": {
+                                "onRun": {
+                                    "command": "{{ Task.File.whoami }}",
+                                },
+                            },
+                        },
+                    },
+                ],
+            },
+        )
+        return job
+
+    def test_no_user_override(
+        self,
+        deadline_resources,
+        deadline_client: DeadlineClient,
+        class_worker: EC2InstanceWorker,
+        job_run_as_user: PosixSessionUser,
+    ) -> None:
+        # WHEN
+        job = self.submit_whoami_job(
+            "No user override",
+            deadline_client,
+            deadline_resources.farm,
+            deadline_resources.queue_a,
+        )
+
+        # THEN
+        job.wait_until_complete(client=deadline_client, max_retries=20)
+
+        job.assert_single_task_log_contains(
+            deadline_client=deadline_client,
+            logs_client=boto3.client(
+                "logs",
+                config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+            ),
+            expected_pattern=rf"I am: {re.escape(job_run_as_user.user)}",
+        )
+
+        assert job.task_run_status == TaskStatus.SUCCEEDED
+
+    def test_config_file_user_override(
+        self,
+        deadline_resources,
+        class_worker: EC2InstanceWorker,
+        deadline_client: DeadlineClient,
+    ) -> None:
+
+        class_worker.stop_worker_service()
+
+        @backoff.on_exception(
+            backoff.constant,
+            Exception,
+            max_time=45,
+            interval=5,
+        )
+        def check_worker_service_stopped() -> None:
+            worker_status_cmd_response = class_worker.send_command(
+                "systemctl is-active deadline-worker"
+            )
+
+            assert worker_status_cmd_response.exit_code != 0
+            assert worker_status_cmd_response.stdout != "active"
+
+        check_worker_service_stopped()
+
+        cmd_result = class_worker.send_command(
+            'sed -i \'s/# posix_job_user = "user:group"/posix_job_user = "config-override:job-override-group"/g\' /etc/amazon/deadline/worker.toml'
+        )
```

Comment:
Follow-up from my comment about the `posix_job_users` fixture, for example this code could use the user fixture directly rather than assuming the user/group exists on the worker
```suggestion
            f'sed -i \'s/# posix_job_user = "user:group"/posix_job_user = "{override_user.user}:{override_user.group}"/g\' /etc/amazon/deadline/worker.toml'
        )
```